Modules/posixmodule.c

```diff
@@ -14962,6 +14962,349 @@ os_waitstatus_to_exitcode_impl(PyObject *module, PyObject *status_obj)
 #endif
 
 
+#if defined(MS_WINDOWS)
+/*[clinic input]
+os._isdir
+
+    path: 'O'
+
+Return True if path is an existing directory.
+
+This follows symbolic links, so both islink() and isdir() can be true for the
+same path.
+
+[clinic start generated code]*/
+
+static PyObject *
+os__isdir_impl(PyObject *module, PyObject *path)
+/*[clinic end generated code: output=e87264f004b1aa44 input=2b77ce7f601eb6bc]*/
+{
+    HANDLE hfile;
+    BOOL close_file = TRUE;
+    FILE_BASIC_INFO info;
+    path_t _path = PATH_T_INITIALIZE("isdir", "path", 0, 1);
+    STRUCT_STAT st;
+    DWORD error;
+    int result;
+
+    if (!path_converter(path, &_path)) {
+        path_cleanup(&_path);
+        if (PyErr_ExceptionMatches(PyExc_ValueError)) {
+            PyErr_Clear();
+            Py_RETURN_FALSE;
+        }
+        else {
+            return NULL;
+        }
```

Comment:
There's a lot of unneeded `else`-`return` clauses, like this one. Would you mind updating the PR with unneeded `else` clauses removed?

```suggestion
        return NULL;
```